Filter parameters from 'application/json' content-type POST requests

marco-santamaria opened the pull request.

IvanMalison commented:
test?

IvanMalison commented on vcr/filters.py:
```diff
@@ -31,13 +32,20 @@ def remove_query_parameters(request, query_parameters_to_remove):
-        request.body = b'&'.join(
-            b'='.join([k, v])
-            for k, vals in post_data.items() for v in vals)
+        try:
```
Why not look at the content type header instead of just try excepting?

marco-santamaria commented:
OK, yesterday I was in a hurry :)
I added some tests and removed the try/except.
Please note that with JSON post data it doesn't look a good idea to me accepting duplicate keys.
Also, with the proposed implementation the order of the keys can change: I don't think it could be a problem.

marco-santamaria commented:
Yes that was the reason. I could have rewritten with dict, but is already merged. Thanks
